# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/mips64/Lowering-mips64.cpp
# Commit: fd8adea7aa17
# Full Hash: fd8adea7aa17b82a6d48c72836823b11ae4c7c66
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 09:30:56
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.2: Establish dependency from tls for x86 callWithABI div/mod i64. r=lth
#   
#   x86 has few register so to do div/mod for i64 it call the runtime
#   and clobber almost all gp registers including WasmTlsReg.
#   To be able to call c++ runtime via Builtin thunk we need to set up WasmTlsReg.
# ==============================================================================

diff -r 6d91f5b4cc32 -r fd8adea7aa17 js/src/jit/mips64/Lowering-mips64.cpp
--- a/js/src/jit/mips64/Lowering-mips64.cpp	Fri Sep 11 08:32:46 2020 +0300
+++ b/js/src/jit/mips64/Lowering-mips64.cpp	Fri Sep 11 05:40:14 2020 +0000
@@ -43,6 +43,10 @@
   defineInt64(lir, div);
 }
 
+void LIRGeneratorMIPS64::lowerWasmBuiltinDivI64(MWasmBuiltinDivI64* div) {
+  MOZ_CRASH("We don't use runtime div for this architecture");
+}
+
 void LIRGeneratorMIPS64::lowerModI64(MMod* mod) {
   if (mod->isUnsigned()) {
     lowerUModI64(mod);
@@ -54,6 +58,10 @@
   defineInt64(lir, mod);
 }
 
+void LIRGeneratorMIPS64::lowerWasmBuiltinModI64(MWasmBuiltinModI64* mod) {
+  MOZ_CRASH("We don't use runtime mod for this architecture");
+}
+
 void LIRGeneratorMIPS64::lowerUDivI64(MDiv* div) {
   LUDivOrModI64* lir = new (alloc())
       LUDivOrModI64(useRegister(div->lhs()), useRegister(div->rhs()), temp());